Source/MLX/MLXArray.swift

```diff
@@ -83,6 +83,42 @@ public final class MLXArray {
         return (0 ..< ndim).map { Int(cShape[$0]) }
     }
 
+    /// Dimensions of the 2d array as a tuple.
+    ///
+    /// ```swift
+    /// let (w, h) = array.shape2
+    /// ```
+    public var shape2: (Int, Int) {
+        let ndim = mlx_array_ndim(ctx)
+        precondition(ndim == 2)
+        let cShape = mlx_array_shape(ctx)!
+        return (Int(cShape[0]), Int(cShape[1]))
+    }
```

Comment:
Similar addition for destructuring `shape`.  I have to add the `2` etc because

```swift
let s = x.shape()
```

is otherwise ambiguous -- is `s` a tuple or an array?

These handle a number of common cases I have run into.